controllers/bsl.go

```diff
@@ -15,22 +15,28 @@ import (
 )
 
 func (r *DPAReconciler) ValidateBackupStorageLocations(dpa oadpv1alpha1.DataProtectionApplication) (bool, error) {
-	if dpa.Spec.Configuration == nil || dpa.Spec.Configuration.Velero == nil {
-		return false, errors.New("DPA CR Velero configuration cannot be nil")
-	}
 	// Ensure we have a BSL or user has specified NoDefaultBackupLocation install
 	if len(dpa.Spec.BackupLocations) == 0 && !dpa.Spec.Configuration.Velero.NoDefaultBackupLocation {
```

Comment:
This one we can remove as well, it is duplication of https://github.com/openshift/oadp-operator/blob/8890679c9b2c6866860badbd3c870ff3392a5ec8/controllers/validator.go#L37-L41

Maybe is even better to bring this part of code https://github.com/openshift/oadp-operator/blob/8890679c9b2c6866860badbd3c870ff3392a5ec8/controllers/validator.go#L30-L41
to here (the other PR was merged @stillalearner :smile: you can rebase now)

Reply:
for maintainers: should we move this code https://github.com/openshift/oadp-operator/blob/8890679c9b2c6866860badbd3c870ff3392a5ec8/controllers/validator.go#L109-L114

up in the validation chain? to do not change the order it was previously?

Reply:
@mateusoliveira43 you are right, actually this backupLocation for loop and volumeSnapshot for loop in the validator.go is also duplicate as it is being checked already in the new functions ensureDataSecretExists and the VSl part also. removing it.

Reply:
done. I also think that bsl and vsl validation should be done first, as the duplicate code we are removing from there was actually doing the same previously, which is now included in ValidateBSL and ValidateVSL functions.